python/pyspark/pandas/groupby.py

```diff
@@ -2356,12 +2356,16 @@ def nunique(self, dropna: bool = True) -> FrameLike:
         Name: value1, dtype: int64
         """
         if dropna:
-            stat_function = lambda col: F.countDistinct(col)
+
+            def stat_function(col: Column) -> Column:
+                return F.countDistinct(col)
```

Comment:
`stat_function = F.countDistinct` should work as well, but I don't have preference here.